packages/@uppy/tus/src/getFingerprint.js

```diff
@@ -0,0 +1,44 @@
+const tus = require('tus-js-client')
+
+function isCordova () {
+  return typeof window !== 'undefined' && (
+    typeof window.PhoneGap !== 'undefined' ||
+    typeof window.Cordova !== 'undefined' ||
+    typeof window.cordova !== 'undefined'
+  )
+}
+
+function isReactNative () {
+  return typeof navigator !== 'undefined' &&
+    typeof navigator.product === 'string' &&
+    navigator.product.toLowerCase() === 'reactnative'
+}
+
+// We override tus fingerprint to uppy’s `file.id`, since the `file.id`
+// now also includes `relativePath` for files added from folders.
+// This means you can add 2 identical files, if one is in folder a,
+// the other in folder b — `a/file.jpg` and `b/file.jpg`, when added
+// together with a folder, will be treated as 2 separate files.
+//
+// For React Native and Cordova, we let tus-js-client’s default
```

Comment:
Is it actually necessary to differentiate between React Native/Cordova and other environments? Is it dangerous if the relative path is just empty there?

Reply:
You are right, it’s not necessary to differentiate, tus-js-client can handle both, but it’s not only about relative path, there’s special handling needed for React Native file id.